command/agent/agent_endpoint.go

```diff
@@ -401,3 +459,27 @@ func (s *HTTPServer) syncChanges() {
 		s.logger.Printf("[ERR] agent: failed to sync changes: %v", err)
 	}
 }
+
+type httpLogHandler struct {
+	filter *logutils.LevelFilter
+	logCh  chan string
+	logger *log.Logger
+}
+
+func (h *httpLogHandler) HandleLog(log string) {
+	// Check the log level
+	if !h.filter.Check([]byte(log)) {
+		return
+	}
+
+	// Do a non-blocking send
+	select {
+	case h.logCh <- log:
+	default:
+		// We can't log synchronously, since we are already being invoked
+		// from the logWriter, and a log will need to invoke Write() which
+		// already holds the lock. We must therefor do the log async, so
+		// as to not deadlock
+		go h.logger.Printf("[WARN] Dropping logs to monitor http endpoint")
```

Comment:
I kind of worry this could make a ton of goroutines if things got into a weird state. I'd just count dropped lines in here, and then kick out a warning up in the agent handler when you are closing out if this count > 0. That way the operator can see that this is going on, but it's super cheap and ok if we are essentially dropping all of the logs.